### PR TITLE
Teacher status change restrictions

### PR DESCRIPTION
**Pull request by 0xf8f8ff**
Any course teacher can now freely promote and demote other course teachers. The course creator can never be demoted.

### CODE REVIEW

File: web/autograder_service.go [hunk 1]
```diff
@@ -217,14 +217,10 @@ func (s *AutograderService) UpdateEnrollment(ctx context.Context, in *pb.Enrollm
 		return nil, status.Errorf(codes.PermissionDenied, "only teachers can update enrollment status")
 	}
 	if s.isCourseCreator(in.CourseID, in.UserID) {
-		s.logger.Error("UpdateEnrollment failed: user is course creator, whose enrollment cannot be changed")
+		s.logger.Errorf("UpdateEnrollment failed: user %s has attempted to demote course creator", usr.GetName())
```
Comment (meling) [inline web/autograder_service.go]: user %s attempted to

Comment (0xf8f8ff) [reply web/autograder_service.go]: Fixed.

File: web/users.go [hunk 2]
```diff
@@ -52,12 +52,17 @@ func (s *AutograderService) getUsers() (*pb.Users, error) {
 // updateUser promotes the given user to administrator and/or
 // make other changes to the user database entry for the given user.
 // The isAdmin flag must be true to promote the given user to admin.
-func (s *AutograderService) updateUser(isAdmin bool, request *pb.User) (*pb.User, error) {
+func (s *AutograderService) updateUser(isAdmin bool, curUser string, request *pb.User) (*pb.User, error) {
```
Comment (meling) [inline web/users.go]: Change signature take `currentUser` (`*pb.User` object) instead. Then replace the `isAdmin` bool and `curUser` string with `currentUser.IsAdmin` and `currentUser.Login`.

File: web/courses.go [hunk 3]
```diff
@@ -38,11 +38,15 @@ func (s *AutograderService) createEnrollment(request *pb.Enrollment) error {
...
 // updateEnrollment accepts or rejects a user to enroll in a course.
-func (s *AutograderService) updateEnrollment(ctx context.Context, sc scm.SCM, request *pb.Enrollment) error {
+func (s *AutograderService) updateEnrollment(ctx context.Context, sc scm.SCM, curUser string, request *pb.Enrollment) error {
 	enrollment, err := s.db.GetEnrollmentByCourseAndUser(request.CourseID, request.UserID)
 	if err != nil {
 		return err
 	}
+	// log changes to teacher status
+	if enrollment.Status == pb.Enrollment_TEACHER || request.Status == pb.Enrollment_TEACHER {
+		s.logger.Debugf("User %s attempts changing enrollment status of user %d from %s to %s", curUser, enrollment.UserID, enrollment.Status, request.Status)
```
Comment (meling) [inline web/courses.go]: I'm still thinking about a way to avoid passing `curUser` just to print a debug statement...

Comment (0xf8f8ff) [reply web/courses.go]: Yeah, I'm not very fond of it either. It is just that we have information about the current user at `autograder_service` level, but no information about the database user/enrollment entry, and vice versa.

**Pull request merged**